paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/DataCatalogTable.java

```diff
@@ -104,27 +124,24 @@ private Schema toSchema(TableSchema tableSchema, Map<String, String> comments) {
                                 builder.withComment(nonPhysicalColumnComments.get(colName));
                             }
                         });
-
         tableSchema
                 .getWatermarkSpecs()
                 .forEach(
                         spec ->
                                 builder.watermark(
-                                        spec.getRowtimeAttribute(), spec.getWatermarkExpr()));
-
+                                        spec.getColumnName(), spec.getWatermarkExpression()));
         if (tableSchema.getPrimaryKey().isPresent()) {
-            UniqueConstraint primaryKey = tableSchema.getPrimaryKey().get();
-            builder.primaryKeyNamed(primaryKey.getName(), primaryKey.getColumns());
+            Schema.UnresolvedPrimaryKey primaryKey = tableSchema.getPrimaryKey().get();
+            builder.primaryKeyNamed(primaryKey.getConstraintName(), primaryKey.getColumnNames());
         }
-
         return builder.build();
     }
 
     @Override
     public CatalogBaseTable copy() {
         return new DataCatalogTable(
                 table,
-                getSchema().copy(),
+                Schema.newBuilder().fromSchema(schema).build(),
```

Comment:
Why need to copy it? I see there is no copy in `DefaultCatalogTable`.

Reply:
The copy was added because the copy had been made in the existing implementation. But I agree with it that the copy is unnecessary. I'll remove it.